blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubPipelineCreateRequest.java

```diff
@@ -179,8 +186,10 @@ public boolean meetsIndexingCriteria() {
                         }
                     });
                 }else {
+                    gitHubSCMNavigator.setPattern(".*");
```

Comment:
Should be moved in the else condition up there at https://github.com/jenkinsci/blueocean-plugin/blob/da60f3b1903de17b5008e60b0e76633a00aaca5d/blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubPipelineCreateRequest.java#L164.

```
                if (sb.length() > 0) {
                    gitHubSCMNavigator.setPattern(sb.toString());
                }else{
                    gitHubSCMNavigator.setPattern(".*");
                }
```
This will also avoid doing extra save you have added in this change. It's optimization of code and will ensure save happens before scheduleBuild() is called  - save happens in `organizationFolder.getNavigators().replace(gitHubSCMNavigator);`. Your code will work fine, its more of optimization that we can do later given its all been tested already. Up to @kzantow as I might be hitting road in couple hours.